### PR TITLE
Begin puppeteer upgrade

### PR DESCRIPTION
**Pull request by varunsub**
Upgraded packages tslint, jest, ts-jest, typescript, @types/jest, puppeteer, @types/puppeteer, and @cliqz/adblocker-puppeteer. Also fixed any breaking changes with this upgrade.

### CODE REVIEW

File: src/key-logging.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const setupKeyLoggingInspector = async (
   page: Page,
   eventDataHandler: (event: BlacklightEvent) => void,
 ) => {
-  page.on("request", (request: Request) => {
```
Comment (simonft) [inline src/key-logging.ts:24]: Why are we removing the type here?

File: __tests__/parser.ts [hunk 2]
```diff
@@ -107,7 +108,7 @@ it("can group fingerprintable window objects", async () => {
   expect(output["NAVIGATOR"]).toEqual(NAVIGATOR_SYMBOLS);
   expect(output["SCREEN"]).toEqual(SCREEN_SYMBOLS);
   expect(output["MEDIA_DEVICES"]).toEqual(MEDIA_DEVICES_SYMBOLS);
-});
+}, 60000);
```
Comment (simonft) [inline __tests__/parser.ts:111]: What's the 60000 here doing? Also the two above.

Comment (varunsub) [reply __tests__/parser.ts:111]: It changes the time limit for a test to run. The default is 5,000 but sometimes it goes over that causing the test case to fail.

File: src/key-logging.ts (hunk 1 above)
Comment (varunsub) [reply src/key-logging.ts:24]: Request type is no longer exported by Puppeteer.

File: __tests__/parser.ts (hunk 2 above)
Comment (simonft) [reply __tests__/parser.ts:111]: milliseconds? @samatt do you have thoughts about a good default here?

Is this something new in the upgrade, or was it like that before?

File: src/key-logging.ts (hunk 1 above)
Comment (samatt) [reply src/key-logging.ts:24]: might be better to replace it with this https://pptr.dev/#?product=Puppeteer&version=v14.1.0&show=api-class-httprequest

File: __tests__/parser.ts (hunk 2 above)
Comment (simonft) [reply __tests__/parser.ts:111]: Oh never mind this is just in __tests__. Never mind.

File: src/key-logging.ts (hunk 1 above)
Comment (varunsub) [reply src/key-logging.ts:24]: Good idea, done!

**Pull request merged**